File: versatIL-frontend/src/model/task/task.ts
export interface TaskModel {
  id: string;
  title: string;
  description: string;
  status: string;
  dueDate: string;
  priority: string;
  progress: number;
  subtasks: SubtaskDetail[];
}

export interface SubtaskDetail {
  id: number;
  title: string;
  completed: boolean;
}

export interface CreateTaskRequest {
  id: string;
  userid: string;
  title: string;
  description: string;
  status: string;
  dueDate: string;
  priority: string;
  progress: number;
  subtasks: string;
}

export interface CreateTaskResponse{
  message: string;
}

export interface GetTasksRequest{
  userid: string;
}

export interface GetTasksResponse{
  messsage: string;
  task: string;
}

export interface DeleteTaskRequest{
  taskid: string;
}

File: versatIL-backend/model/task.go
package model

type GetTaskRequest struct {
	UserId string `json:"userid"`
}

type Task struct {
	Id          string `json:"id"`
	UserId      string `json:"userId"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Status      string `json:"status"`
	DueDate     string `json:"dueDate"`
	Priority    string `json:"priority"`
	Progress    int32  `json:"progress"`
	Subtasks    string `json:"subtasks"`
}

type Subtask struct {
	Id        string `json:"id"`
	Title     string `json:"title"`
	Completed bool   `json:"completed"`
}

type DeleteTaskRequest struct {
	TaskId string `json:"taskid"`
}
